### ISSUE TITLE
Allow users to customize prompt layout

### ISSUE THREAD

**Issue by satayev** [dimo414/prompt.gem#13]
At the moment PS1 is hard-coded env_functions.sh. Modifying it locally is obviously possible, but leads to infrequent merge conflicts.

Would it be possible to expose LAYOUT in base.conf of some sorts?

**Comment by dimo414** [comment]
prompt.gem is intentionally prescriptive about how PS1 gets rendered. Originally the intended solution if someone prefers a different prompt would have been to create their own gem that configures the prompt how they'd like. However over time prompt.gem has become more and more powerful and it would definitely be a shame if someone had to reimplement or fork most of the gem just to tweak the layout.

The right solution long-term might be to split the gem in two, but I'm open to smaller improvements in the meantime. Can you elaborate on how you envision `LAYOUT` working?

**Comment by satayev** [comment]
Makes sense, I'd be okay either way to be honest (perhaps migrate some of the functions into util.gem?). Just to make sure I don't think this is a high priority issue by any means; just a minor inconvenience that results in merge conflicts once in a while.

We could expose available blocks for PS1 (e.g. machine, env, host, dir, etc.) and allow LAYOUT to use them; PS1 is then expanded and set to said layout. (I don't know bash enough, but this sounds doable?)

So, for example, my personal LAYOUT would be:
LAYOUT='\n[${exit_code_display}] [${env} ${short_pwd}]\n${prompt_string}'

**Comment by dimo414** [comment]
One option would be to move the host/pwd rendering into an `ENV_INFO` function, then you could just rearrange that array. Would that work?
